src/SFA.DAS.QnA.Application.UnitTests/CommandsTests/ResetPageAnswersBySectionNoHandlerTests/When_page_found.cs

```diff
@@ -44,11 +44,10 @@ public async Task Then_questiontag_is_reset()
 
             var getApplicationDataResponse = await GetApplicationDataHandler.Handle(new GetApplicationDataRequest(ApplicationId), CancellationToken.None);
 
-            var applicationData = JObject.Parse(getApplicationDataResponse.Value);
+            var applicationData = JsonNode.Parse(getApplicationDataResponse.Value).AsObject();
             var questionTag = applicationData["Q1"];
 
-            questionTag.Should().NotBeNull();
-            questionTag.Value<string>().Should().BeNullOrEmpty();
+            questionTag.Should().BeNull();
```

Comment:
It seems that the spirit of the original test, though poorly conveyed, was to check that `applicationData` contained one property-value pair where the property is `"Q1"` and the value is `null`. You've rewritten it to be in line with the behaviour of Text.Json, but now it's not doing the same checks. We should rewrite this test to preserve the original intent.

I would suggest parsing into a JsonDocument on line 47:
`var applicationData = JsonDocument.Parse(getApplicationDataResponse.Value).RootElement;`

After doing so, `applicationData` will have methods that allow you to check its properties and values. Use those to assert that the property "Q1" exists, and has a value of null.

I imagine this shouldn't involve more than 3-5 lines of code.

Reply:
done

Reply:
Since `questionTag` is a struct, it will never be null, so the assertion on 50 isn't doing anything. I think this test should assert that `applicationData` has only one property, and that its value is null